authentik/stages/authenticator_email/tasks.py

```diff
@@ -0,0 +1,105 @@
+"""email stage tasks"""
+
+from email.utils import make_msgid
+from smtplib import SMTPException
+from typing import Any
+
+from celery import group
+from django.core.mail import EmailMultiAlternatives
+from django.core.mail.utils import DNS_NAME
+from django.utils.text import slugify
+from structlog.stdlib import get_logger
+
+from authentik.events.models import Event, EventAction, TaskStatus
+from authentik.events.system_tasks import SystemTask
+from authentik.root.celery import CELERY_APP
+from authentik.stages.authenticator_email.models import AuthenticatorEmailStage
+from authentik.stages.email.utils import logo_data
+
+LOGGER = get_logger()
+
+
+def send_mails(stage: AuthenticatorEmailStage, *messages: list[EmailMultiAlternatives]):
+    """Wrapper to convert EmailMessage to dict and send it from worker"""
+    tasks = []
+    for message in messages:
+        tasks.append(send_mail.s(message.__dict__, str(stage.pk)))
+    lazy_group = group(*tasks)
+    promise = lazy_group()
+    return promise
+
+
+def get_email_body(email: EmailMultiAlternatives) -> str:
+    """Get the email's body. Will return HTML alt if set, otherwise plain text body"""
+    for alt_content, alt_type in email.alternatives:
+        if alt_type == "text/html":
+            return alt_content
+    return email.body
+
+
+@CELERY_APP.task(
+    bind=True,
+    autoretry_for=(
+        SMTPException,
+        ConnectionError,
+        OSError,
+    ),
+    retry_backoff=True,
+    base=SystemTask,
+)
+def send_mail(self: SystemTask, message: dict[Any, Any], email_stage_pk: str | None = None):
```

Comment:
this we shouldn't need to duplicate, it should be possible to re-use the email stage tasks (even if the authenticator stage is slightly different, it still has the same signature for the things relevant here)